VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $organization,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $organizationUnit,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $stateOrProvince,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $email
+    )
+
+    if (Test-VCFConnection -server $server) {
+        if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+            $domainType = Get-VCFWorkloadDomain -name $workloadDomain
+            $resourcesObject = gatherSddcInventory -domainType $domainType.type -workloadDomain $workloadDomain
+            Write-Output "Populating $($workloadDomain)-requestCsrSpec.json"
+            $path = Join-Path $pwd ""
+            $csrGenerationSpecJson =
+            '{
+            "csrGenerationSpec": {
+                "country": "'+ $country + '",
+                "email": "'+ $email + '",
+                "keyAlgorithm": "'+ "RSA" + '",
+                "keySize": "'+ "2048" + '",
+                "locality": "'+ $locality + '",
+                "organization": "'+ $organization + '",
+                "organizationUnit": "'+ $organizationUnit + '",
+                "state": "'+ $stateOrProvince + '"
+                },
+            '
+            $resourcesBodyObject += [pscustomobject]@{
+                resources = $resourcesObject
+            }
+            $resourcesBodyObject | ConvertTo-Json -Depth 10 | Out-File -FilePath $path"temp.json"
+            Get-Content $path"temp.json" | Select-Object -Skip 1 | Set-Content $path"temp1.json"
+            $resouresJson = Get-Content $path"temp1.json" -Raw
+            Remove-Item -Path $path"temp.json"
+            Remove-Item -Path $path"temp1.json"
+            $requestCsrSpecJson = $csrGenerationSpecJson + $resouresJson
+            $requestCsrSpecJson | Out-File $path"$($workloadDomain)-requestCsrSpec.json"
+            Write-Output "Requesting CSRs for Components Associated with Domain $($workloadDomain)"
+            $myTask = Request-VCFCertificateCSR -domainName $($workloadDomain) -json $path"$($workloadDomain)-requestCsrSpec.json"
+            Do {
+                Write-Output "Waiting for CSRs to be Created for Components Associated with Domain $($workloadDomain)"
+                Start-Sleep 6
+                $response = Get-VCFTask $myTask.id
+            } While ($response.status -eq "IN_PROGRESS")
+            If ($response.status -eq "FAILED") {
+                Write-Output "Workflow Completed with status $($response.status)" 
+            }
+            elseIf ($response.status -eq "SUCCESSFUL") {
+                Write-Output "Workflow Completed with status $($response.status)"
+            } else {
+                Write-Output "Workflow Completed with an unrecognized status of $($response.status). Please check before proceeding."
```

Comment:
```suggestion
                Write-Warning "Workflow completed with an unrecognized status: $($response.status). Please check status before proceeding."
```

Reply:
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)